### ISSUE TITLE
Button Control Invokes :pressed Pseudoclass but not Click

### ISSUE THREAD

**Issue by robloo** [AvaloniaUI/Avalonia#7626]
**Describe the bug**

There is a strange behavior in the `SplitButton`/`ToggleSplitButton` that isn't related to the `SplitButton` itself. The internal `Button` control is applying the :pressed Pseudoclass on click but not raising the Click event. This means the style state does not match the actual button state.

This happens with both the primary and secondary buttons in the last pixel (Border area) as shown below.

![152631176-2ca1f14a-892a-4678-880b-66d4646cf81c](https://user-images.githubusercontent.com/17993847/154331006-076c5b6a-8615-46ce-9d20-3327bd37f782.gif)

According to the discussion in the SplitButton PR, @maxkatz6 Advised that hit testing seems to not work correctly here https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Button.cs#L363

Interestingly, this does not seem to be an issue with Button itself so this may be a more complex issue related to using Button as a child in another control's template.

**To Reproduce**

Click on the last pixel just below the SplitButton in the ConrolCatalog.

**Expected behavior**

**Screenshots**

![152631176-2ca1f14a-892a-4678-880b-66d4646cf81c](https://user-images.githubusercontent.com/17993847/154331006-076c5b6a-8615-46ce-9d20-3327bd37f782.gif)

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 0.10.12 (dev master branch around January 2022).

**Additional context**

**Comment by robloo** [comment]
This "works" now after the ControlTheme rewrite of Fluent. It works because the button animation wasn't added back to the SplitButton. So this proves the theory in #8550 is correct and this is a side-effect of the animation matrix transform and hit testing sequence.

**Comment by timunie** [comment]
@robloo #13797 seems to have this issue solved as well. At least in a simple tests I did I cannot reproduce. If you also think the same, I guess we can close this issue.

**Comment by robloo** [comment]
I'll try to remember to re-test this soon myself.

**Comment by robloo** [comment]
@timunie I still see this issue.

Windows 10 Pro 22H2
Avalonia 11.0.6+ (36e7ab50043a871d725f72be5366950d93737801)

![issue](https://github.com/AvaloniaUI/Avalonia/assets/17993847/41c60c03-fd03-451f-9460-b90814cd7ed0)

**Comment by timunie** [comment]
Looks like this record is from FA sample? Because using FluentTheme I wasn't able to reproduce. Need to check that on my own. Maybe in DevTools I find a hint then.

**Comment by robloo** [comment]
@timunie 

> Looks like this record is from FA sample?

No, it's pure Avalonia code at latest master running in ControlCatalog.NetCore with FluentTheme dark. It is a little tricky to see because you have to get it to within 1 pixel. That might also relate to any display scaling you have (I'm at 150%).

**Comment by memorydream** [comment]
I tried test it , problem seems in CompositingRenderer

First , CompositingRenderer copy Visual properties to CompositionVisual , but it will calculate final RenderTransform matrix
https://github.com/AvaloniaUI/Avalonia/blob/2a7b5568b7f61fe6df9c46a05bf1c9d474c9290e/src/Avalonia.Base/Visual.Composition.cs#L156-L163

And do hit test in CompositionTarget
https://github.com/AvaloniaUI/Avalonia/blob/2a7b5568b7f61fe6df9c46a05bf1c9d474c9290e/src/Avalonia.Base/Rendering/Composition/CompositionTarget.cs#L96-L102

https://github.com/AvaloniaUI/Avalonia/blob/2a7b5568b7f61fe6df9c46a05bf1c9d474c9290e/src/Avalonia.Base/Rendering/Composition/CompositionTarget.cs#L75-L85

Hit point is calculated in `TryTransformTo` 

`visual.Size` come from `Bounds.Width` and `Bounds.Height` , but bound size doesn't change with RenderTransfrom from pseudo classes (or Transitions ? i didn't test it) , value also has not change in DevTools

In  Button (`<Setter Property="RenderTransform" Value="scale(0.98)" />`) , when pointer release in edge of control , point position actual greater than control bounds

So , we got the bug , OnPointerReleased always called, but hit test failed in CompositingRenderer

---
hit test entry is `GetVisualsAt`
https://github.com/AvaloniaUI/Avalonia/blob/2a7b5568b7f61fe6df9c46a05bf1c9d474c9290e/src/Avalonia.Controls/Button.cs#L413-L417

**Comment by robloo** [comment]
@memorydream Wow! Thanks so much for taking the time to dig into this! I'm sure a PR will be appreciated if you have the time.

**Comment by memorydream** [comment]
> @memorydream Wow! Thanks so much for taking the time to dig into this! I'm sure a PR will be appreciated if you have the time.

Sorry , I don't know how to do is right , they are too low-level
Maybe core team can give help ?

**Comment by memorydream** [comment]
An idea ... I haven't try to implement it  
If add new dependency properties `VisualTransform` and `VisualTransfromOrigin` , its only for render transform matrix calculation , not change `RenderTransfrom`

behaviour after add VisualTransform:

render matrix = RenderTransform * VisualTransform
hit test matrix = RenderTransfrom

But should we have two similar properties for some slightly difference use cases ? and are `RenderTransform` and `VisualTransform` confusing ? or wait refactor animation ?

**Comment by memorydream** [comment]
cc @kekekeks
